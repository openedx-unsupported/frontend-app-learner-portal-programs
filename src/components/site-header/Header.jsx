import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import Responsive from 'react-responsive';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';
import {
  ensureConfig,
} from '@edx/frontend-platform';

import DesktopHeader from './DesktopHeader';
import MobileHeader from './MobileHeader';

import './Header.scss';

ensureConfig([
  'LMS_BASE_URL',
  'LOGOUT_URL',
  'LOGIN_URL',
], 'Header component');

function Header({
  headerLogo, logoAltText, logoDestination, userMenu, mainMenu,
}) {
  const { authenticatedUser } = useContext(AppContext);

  const props = {
    logo: headerLogo,
    logoAltText,
    logoDestination,
    loggedIn: authenticatedUser !== null,
    username: authenticatedUser ? authenticatedUser.username : null,
    avatar: authenticatedUser && authenticatedUser.profileImage
      ? authenticatedUser.profileImage.imageUrlMedium
      : null,
    mainMenu,
    userMenu,
  };

  return (
    <>
      <Responsive maxWidth={768}>
        <MobileHeader {...props} />
      </Responsive>
      <Responsive minWidth={769}>
        <DesktopHeader {...props} />
      </Responsive>
    </>
  );
}

Header.propTypes = {
  intl: intlShape.isRequired,
  headerLogo: PropTypes.string,
  logoAltText: PropTypes.string,
  logoDestination: PropTypes.string,
  mainMenu: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.array,
  ]),
  userMenu: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.oneOf(['item', 'menu']),
    href: PropTypes.string,
    content: PropTypes.string,
  })),
};

Header.defaultProps = {
  headerLogo: null,
  logoAltText: null,
  logoDestination: null,
  mainMenu: [],
  userMenu: [],
};

export default injectIntl(Header);
