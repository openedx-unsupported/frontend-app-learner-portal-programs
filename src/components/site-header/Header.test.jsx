/* eslint-disable react/jsx-no-constructed-context-values */
import React from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import renderer from 'react-test-renderer';
import { Context as ResponsiveContext } from 'react-responsive';
import { AppContext } from '@edx/frontend-platform/react';

import Header from './Header';

describe('<Header />', () => {
  const initialProps = {
    headerLogo: 'logo.jpg',
    logoDestination: 'mysite.com',
    logoAltText: 'mysite',
    userMenu: [
      {
        type: 'item',
        href: process.env.LMS_BASE_URL,
        content: 'Dashboard',
      },
      {
        type: 'item',
        href: '/',
        content: 'My Masters Degree',
      },
    ],
  };

  it('renders correctly for authenticated desktop', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 1280 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider
            value={{
              authenticatedUser: {
                userId: 'abc123',
                username: 'edX',
                roles: [],
                administrator: false,
              },
              config: {
                LMS_BASE_URL: process.env.LMS_BASE_URL,
                SITE_NAME: process.env.SITE_NAME,
                LOGIN_URL: process.env.LOGIN_URL,
                LOGOUT_URL: process.env.LOGOUT_URL,
                LOGO_URL: process.env.LOGO_URL,
              },
            }}
          >
            <Header {...initialProps} />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = renderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for authenticated mobile', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 500 }}>
        <IntlProvider locale="en" messages={{}}>
          <AppContext.Provider
            value={{
              authenticatedUser: {
                userId: 'abc123',
                username: 'edX',
                roles: [],
                administrator: false,
              },
              config: {
                LMS_BASE_URL: process.env.LMS_BASE_URL,
                SITE_NAME: process.env.SITE_NAME,
                LOGIN_URL: process.env.LOGIN_URL,
                LOGOUT_URL: process.env.LOGOUT_URL,
                LOGO_URL: process.env.LOGO_URL,
              },
            }}
          >
            <Header {...initialProps} />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const wrapper = renderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
