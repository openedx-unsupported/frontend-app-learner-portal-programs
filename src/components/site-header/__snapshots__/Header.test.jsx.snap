// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly for authenticated desktop 1`] = `
<div>
  <header
    class="site-header-desktop"
  >
    <a
      class="nav-skip sr-only sr-only-focusable"
      href="#main"
    >
      Skip to main content
    </a>
    <div
      class="container-fluid null"
    >
      <div
        class="nav-container position-relative d-flex align-items-center"
      >
        <a
          class="logo"
          href="mysite.com"
        >
          <img
            alt="mysite"
            class="d-block"
            src="logo.jpg"
          />
        </a>
        <nav
          aria-label="Main"
          class="nav main-nav"
        />
        <nav
          aria-label="Secondary"
          class="nav secondary-menu-container align-items-center ml-auto"
        >
          <div
            class="menu null"
          >
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              aria-label="Account menu for edX"
              class="menu-trigger btn btn-outline-primary d-inline-flex align-items-center pl-2 pr-3"
            >
              <span
                class="avatar overflow-hidden d-inline-flex rounded-circle mr-2"
                style="height: 1.5em; width: 1.5em;"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  height="24px"
                  role="img"
                  style="width: 1.5em; height: 1.5em;"
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M4.10255106,18.1351061 C4.7170266,16.0581859 8.01891846,14.4720277 12,14.4720277 C15.9810815,14.4720277 19.2829734,16.0581859 19.8974489,18.1351061 C21.215206,16.4412566 22,14.3122775 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,14.3122775 2.78479405,16.4412566 4.10255106,18.1351061 Z M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M12,13 C9.790861,13 8,11.209139 8,9 C8,6.790861 9.790861,5 12,5 C14.209139,5 16,6.790861 16,9 C16,11.209139 14.209139,13 12,13 Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              edX
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> renders correctly for authenticated mobile 1`] = `
<div>
  <header
    aria-label="Main"
    class="site-header-mobile d-flex justify-content-between align-items-center shadow sticky-top"
  >
    <a
      class="nav-skip sr-only sr-only-focusable"
      href="#main"
    >
      Skip to main content
    </a>
    <div
      class="w-100 d-flex justify-content-center"
    >
      <a
        class="logo"
        href="mysite.com"
        itemtype="http://schema.org/Organization"
      >
        <img
          alt="mysite"
          class="d-block"
          src="logo.jpg"
        />
      </a>
    </div>
    <div
      class="w-100 d-flex justify-content-end align-items-center"
    >
      <nav
        aria-label="Secondary"
        class="menu position-static"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          aria-label="Account Menu"
          class="menu-trigger icon-button"
          title="Account Menu"
        >
          <span
            class="avatar overflow-hidden d-inline-flex rounded-circle null"
            style="height: 1.5rem; width: 1.5rem;"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24px"
              role="img"
              style="width: 1.5rem; height: 1.5rem;"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px"
            >
              <path
                d="M4.10255106,18.1351061 C4.7170266,16.0581859 8.01891846,14.4720277 12,14.4720277 C15.9810815,14.4720277 19.2829734,16.0581859 19.8974489,18.1351061 C21.215206,16.4412566 22,14.3122775 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,14.3122775 2.78479405,16.4412566 4.10255106,18.1351061 Z M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M12,13 C9.790861,13 8,11.209139 8,9 C8,6.790861 9.790861,5 12,5 C14.209139,5 16,6.790861 16,9 C16,11.209139 14.209139,13 12,13 Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </nav>
    </div>
  </header>
</div>
`;
